### PR TITLE
Add a small blurb before each utility doc section

### PR DESCRIPTION
**Pull request by rohan-varma**
#### Context
- This should help provide a brief overview of what the specific utilities enable / should be used for. It's a little abrupt IMO to cut straight to the documentation of the utilities, so having this brief description should be helpful.

#### Changelog
- ...

#### Test plan
- 
image

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -97,6 +104,7 @@ TorchTune tutorials.
 
    examples/finetune_llm
    examples/lora_finetune
+   examples/qlora_finetune
```
Comment (RdoubleA) [inline docs/source/index.rst:107]: this is not in yet?

Comment (rohan-varma) [reply docs/source/index.rst]: Oops! Yeah some code from my other PR snuck in

File: docs/source/api_ref_utilities.rst [hunk 2]
```diff
@@ -10,6 +10,9 @@ torchtune.utils
 Checkpointing
 -------------
 
+TorchTune offers checkpointers to allow users to seamlessly transition between checkpoint formats for training and interoperate with the rest of the ecosystem. For a comprehensive overview of
```
Comment (ebsmothers) [inline docs/source/api_ref_utilities.rst:13]:
```suggestion
TorchTune offers checkpointers to allow seamless transitioning between checkpoint formats for training and interoperability with the rest of the ecosystem. For a comprehensive overview of
```

**Review by RdoubleA — APPROVED** [review]

**Review by ebsmothers — APPROVED** [review]
one nit, otherwise looks good!

**Pull request merged**